docs/metrics/bleu.md

```diff
@@ -0,0 +1,123 @@
+
+## Implementation Details
+### Definition
+The BLEU score is mathematically defined as:
+
+$$\begin{align*} \text{BLEU} &= \text{Brevity Penalty} \times \text{n-Gram Overlap} \\
+&= \min\left(1, \exp\left(1 - \frac{\text{reference length}}{\text{output length}}\right)\right) \times \left(\prod_{i=1}^{4}\text{i-Gram Precision}^{1/4}\right)
+\end{align*}$$
+
+where the i-Gram precision is calculated as:
+
+$$
+p_i = \frac{\text{Clipped} \text{ count of matching i-grams in machine-generated text}^1}{\text{Total number of i-grams in machine-generated text}}
+$$
+
+Below, we explain the significance of the two components in the BLEU formula.
+??? info "n-Gram Overlap"
+    The **n-Gram Overlap** counts the number of 1-grams, 2-grams, 3-grams, and 4-grams of the output text that match the 1-, ..., 4-grams in the reference text — which is analogous to a precision score for the text. The 1-gram precision ensures that the correct vocabulary is used, whereas the 4-gram precision ensures that the generated text is coherent.
+??? info "Brevity Penalty"
+    The **Brevity Penalty** is also applied to penalize the score for generating sentences that are less in length than the reference text. This is due to the fact that the n-Gram Overlap precision tends to give disproportionately high values to generated texts that are very short in length, but contain most of the n-grams in the reference text.
+
+<div class="footnote-content">
+    <p style="font-size: smaller;">
+        <sup>1</sup> The <i>clipped count of matching i-grams in machine-generated text</i> is the minimum between the count of i-grams in the machine-generated text and the maximum count of i-grams in any of the reference texts for a given i-gram.
+    </p>
+</div>
+
+### Interpretation
+A known fact about BLEU scores is that they are not to be compared between different workflows and tasks. An excellent BLEU score for one task may be subpar for another. The score only serves as a general guideline to quantify the performance of your model — not to replace human judgement. That being said, what do these scores really mean, and how can we decipher them?
+
+1. **Higher is better**: Though BLEU scores may vary among tasks, one thing is for sure - higher scores are better. Generally speaking, a commonly accepted guideline is as follows: <br>
+
+    | Score | Qualitative Interpretation |
+    | ---   | ---                        |
+    | < 0.1 | Useless output             |
+    | 0.1 - 0.4 | Varies in quality; May not be acceptable |
+    | 0.4 - 0.6 | High quality generated text |
+    | > 0.6 | Better than human quality  |
```

Comment:
> Better than human quality

Is this true?

Reply:
Yup, this is true! https://cloud.google.com/translate/automl/docs/evaluate
<img width="582" alt="image" src="https://github.com/kolenaIO/kolena/assets/31229051/98964d4b-0dde-406d-a855-1f7b516563d7">